### DISCUSSION TITLE
Process to generate an ansible role from this repo

### DISCUSSION THREAD

**Discussion by jseiser** [ComplianceAsCode/content#11495] [Q&A] [answered]
Is there a documented process to generate an ansible role from the content in this repo?  We use https://github.com/RedHatOfficial/ansible-role-rhel8-stig to STIG our AMI's, but it is rarely kept up to date.  I assume we could just generate the role and run it ourselves.

Thanks,

**Comment by Mab879** [comment]
Sorry for the late reply.

These roles are created by the `utils/ansible_playbook_to_role.py` script. You will want to pass the `--dry-run` flag to the script when you run it.

1. Clone the repo
2. Install the [Build Dependencies](https://complianceascode.readthedocs.io/en/latest/manual/developer/02_building_complianceascode.html#installing-build-dependencies) including PyGithub)   
3. `cd content`
3. `./build_product -j$(nproc)  rhel8 rhel9 rhel7` (you can remove products in this list to your needs)
4. `PYTHONPATH="." ./utils/ansible_playbook_to_role.py --dry-run out --build-playbooks-dir ./build/ansible`
5. All roles should be in the out folder.

Hope this helps.


All of the playbooks will be in `out`.

**Reply by Mab879** [reply, marked as answer]
I created https://github.com/ComplianceAsCode/content/pull/11747 to better document these steps.
